### PR TITLE
refactor(console): refactor language options and refine documentation l10n url logic

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

* Refactored language options used in both `@logto/phrases` and `@logto/phrases-ui`
* Added supported documentation languages so that unsupported user locale will fall back to English on clicking the "Docs" link in Admin Console

## Testing

- [x] Tested only Chinese language will end up appending zh-CN in the docusaurus URL, others remain using English.

### CODE REVIEW

File: packages/console/src/components/AppContent/components/Sidebar/hook.tsx [hunk 1]
```diff
@@ -111,7 +112,7 @@ export const useSidebarMenuItems = (): {
           Icon: Document,
           title: 'docs',
           externalLink: `https://docs.logto.io/${conditionalString(
-            language !== Language.English && language.toLowerCase()
+            language === Language.Chinese && language.toLowerCase()
```
Comment (gao-sun) [inline packages/console/src/components/AppContent/components/Sidebar/hook.tsx:115]: will be better to maintain a l10n set of docs

File: packages/console/src/components/AppContent/components/Sidebar/hook.tsx [hunk 2]
```diff
@@ -48,7 +47,9 @@ export const useSidebarMenuItems = (): {
   const {
     data: { getStartedHidden },
   } = useUserPreferences();
-  const language = useLanguage();
+  const {
+    i18n: { language },
+  } = useTranslation();
```
Comment (gao-sun) [inline packages/console/src/components/AppContent/components/Sidebar/hook.tsx:52]: so `useLanguage` is not needed anymore?

Comment (demonzoo) [reply packages/console/src/components/AppContent/components/Sidebar/hook.tsx:52]: You know what... I think these late night code need to be refactored. I take a second look at these changes and now have the same question for myself. 😂

Comment (gao-sun) [reply packages/console/src/components/AppContent/components/Sidebar/hook.tsx:52]: lol

File: packages/console/src/components/AppContent/components/Sidebar/hook.tsx (hunk 1 above)
Comment (demonzoo) [reply packages/console/src/components/AppContent/components/Sidebar/hook.tsx:115]: Oh, I see. So right now the new set of docs languages will only contain English and Chinese

File: packages/phrases-ui/src/types.ts [hunk 3]
```diff
@@ -12,9 +12,13 @@ export enum Language {
...
 
-export const languageOptions = [
-  { value: Language.English, title: 'English' },
-  { value: Language.Chinese, title: '中文' },
-  { value: Language.Turkish, title: 'Turkish' },
-  { value: Language.Korean, title: '한국어' },
-];
+const languageCodeAndDisplayNameMappings: Record<Language, string> = {
+  [Language.English]: 'English',
+  [Language.Chinese]: '中文',
```
Comment (gao-sun) [inline packages/phrases-ui/src/types.ts:17]:
```suggestion
  [Language.Chinese]: '简体中文',
```

File: packages/phrases/src/types.ts [hunk 4]
```diff
@@ -19,11 +19,15 @@ export enum Language {
...
 
-export const languageOptions = [
-  { value: Language.English, title: 'English' },
-  { value: Language.Chinese, title: '中文' },
-  { value: Language.Turkish, title: 'Türkçe' },
-  { value: Language.Korean, title: '한국어' },
-];
+const languageCodeAndDisplayNameMappings: Record<Language, string> = {
+  [Language.English]: 'English',
+  [Language.Chinese]: '中文',
```
Comment (gao-sun) [inline packages/phrases/src/types.ts:24]:
```suggestion
  [Language.Chinese]: '简体中文',
```

File: packages/console/src/hooks/use-documentation-url.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+const documentationSiteUrl = 'https://docs.logto.io';
+
+const useDocumentationUrl = () => {
+  const {
+    i18n: { language },
+  } = useTranslation();
+
+  const documentationLink = Object.values<string>(DocumentationLanguage)
```
Comment (gao-sun) [inline packages/console/src/hooks/use-documentation-url.ts:18]:
```suggestion
  const documentationUrl = Object.values<string>(DocumentationLanguage)
```

File: packages/console/src/hooks/use-documentation-url.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    .filter((language) => language !== DocumentationLanguage.English)
+    .includes(language)
+    ? `${documentationSiteUrl}/${language.toLocaleLowerCase()}`
+    : documentationSiteUrl;
+
+  return documentationLink;
```
Comment (gao-sun) [inline packages/console/src/hooks/use-documentation-url.ts:24]:
```suggestion
  return documentationUrl;
```

**Review by gao-sun — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

**Pull request merged**